src/sinks/nats.rs

```diff
@@ -186,29 +194,33 @@ impl StreamSink<Event> for NatsSink {
 
             let event_byte_size = event.size_of();
 
+            let finalizers = event.take_finalizers();
             let mut bytes = BytesMut::new();
-            if self.encoder.encode(event, &mut bytes).is_err() {
-                // Error is logged by `Encoder`.
+            self.encoder.encode(event, &mut bytes).map_err(|_| {
+                // Error is handled by `Encoder`.
+                finalizers.update_status(EventStatus::Errored);
                 continue;
```

Comment:
I think you need to put this back as `is_err()` and `continue;` outside of a closure.